src/new_storage/storage_manager/handle.rs

```diff
@@ -414,80 +338,272 @@ impl FileHandleInner {
+        self: Arc<Self>,
+        mut write_back_receiver: Receiver<Task>,
+        mut close_receiver: Receiver<()>,
+        metadata_client: Arc<M>,
+    ) {
+        //  Create a timer to flush the cache every 200ms.
+        let mut interval = tokio::time::interval(std::time::Duration::from_millis(200));
+        let mut tasks = Vec::new();
+        let mut result = None;
+        loop {
+            tokio::select! {
+                Some(()) = close_receiver.recv() => {
+                    info!("The {:?} write back task is closed.", self.ino);
+                    return;
+                }
+                Some(task) = write_back_receiver.recv() => {
+                    match task {
+                        Task::Pending(task) => {
+                            info!("Get pending Write back the block. ino: {:?}", self.ino);
+                            tasks.push(task);
+                            if tasks.len() >= 10 {
+                                let slf_clone = Arc::clone(&self);
+                                let res = slf_clone.write_blocks(&tasks, Arc::clone(&metadata_client)).await;
+                                if let Some(e) = res {
+                                    result.get_or_insert(e);
+                                }
+                                tasks.clear();
+                            }
+                        }
+                        Task::Flush(tx) => {
+                            info!("Flush the cache. ino: {:?}", self.ino);
+                            let slf_clone = Arc::clone(&self);
+                            let res = slf_clone.write_blocks(&tasks, Arc::clone(&metadata_client)).await;
+                            tasks.clear();
+                            let res = result.take().or(res);
+                            if let Err(Some(e)) = tx.send(res) {
+                                error!("Failed to send storage error back to `Writer`, the error is {e}.");
+                            }
+                        }
+                        Task::Finish(tx) => {
+                            info!("Finish the write back task. ino: {:?}", self.ino);
+                            let slf_clone = Arc::clone(&self);
+                            let res = slf_clone.write_blocks(&tasks, Arc::clone(&metadata_client)).await;
+                            tasks.clear();
+                            let res = result.take().or(res);
+                            if let Err(Some(e)) = tx.send(res) {
+                                error!("Failed to send storage error back to `Writer`, the error is {e}.");
+                            }
+
+                            // Check last write task size with mem attr
+                            return;
+                        }
+                    }
+                }
+                _ = interval.tick() => {
+                    let slf_clone = Arc::clone(&self);
+                    slf_clone.write_blocks(&tasks, Arc::clone(&metadata_client)).await;
```

Comment:
If len(tasks)==0, should not call write_blocks

Reply:
OK, I find that `write_blocks` will quickly return if the task is none, so we can keep it.